diesel/src/types/mod.rs

```diff
@@ -1,5 +1,17 @@
 //! Types which represent a native SQL data type, and the conversions between
-//! them and Rust primitives. Additional types can be added by other crates.
+//! them and Rust primitives. The structs in this module are *only* used as
+//! markers to represent a SQL type, and shouldn't be used in your structs. See
+//! the documentation for each type to see the Rust types that can be used with
+//! a corresponding SQL type. Additional types can be added by other crates.
```

Comment:
This might be a good place to mention one other crate that does this, as people reading this might want to implement types themselves and having example code to look at might be nice.